dns/config.go

```diff
@@ -14,6 +16,8 @@ type Config struct {
 	keyname   string
 	keyalgo   string
 	keysecret string
+	protocol  string
+	timeout   int
```

Comment:
You could consider using `time.ParseDuration()` here to give a friendly input such as `30s`. At the very least change it to `timeout_seconds` so the units are clear.